src/base/perf_monitor.c

```diff
@@ -4699,10 +4701,68 @@ perfmon_print_timer_to_buffer (char **s, int stat_index, UINT64 * stats_ptr, int
 }
 
 // *INDENT-OFF*
+//////////////////////////////////////////////////////////////////////////
+// thread workers section
+//////////////////////////////////////////////////////////////////////////
+
+// note - current monitor implementation requires a single point to manage all statistics meta information (stats
+//        count and names). this prevents us from isolating statistics management per each module, because some
+//        modules are restricted to server only, while the monitor implementation is same for server and client.
+//
+//        ideally, meta information should be fetched from server and not necessarily known from start by client.
+//        this would prevent any miss-matches between server and clients and better module encapsulation. performance
+//        monitor would only store the values and collect from all modules rather than keeping extended information
+//        on each module (like it does now with page buffer, thread and others)
+//
+//        for now, we are forced to duplicate here some information about thread workers and daemons that is normally
+//        restricted to server
+//
+
+// NOTE - should match cubthread::Worker_pool_statdef
+const char *perfmon_Portable_worker_stat_names [] =
+  {
+    "Counter_start_thread",
+    "Timer_start_thread",
+    "Counter_create_context",
+    "Timer_create_context",
+    "Counter_execute_task",
+    "Timer_execute_task",
+    "Counter_retire_task",
+    "Timer_retire_task",
+    "Counter_found_task_in_queue",
+    "Timer_found_task_in_queue",
+    "Counter_wakeup_with_task",
+    "Timer_wakeup_with_task",
+    "Counter_recycle_context",
+    "Timer_recycle_context",
+    "Counter_retire_context",
+    "Timer_retire_context"
+  };
+static const size_t PERFMON_PORTABLE_WORKER_STAT_COUNT =
+  sizeof (perfmon_Portable_worker_stat_names) / sizeof (const char *);
+
 static size_t
 thread_stats_count (void)
 {
-  return cubthread::wpstat::STATS_COUNT * 2;
+#if defined (SERVER_MODE)
+  assert (PERFMON_PORTABLE_WORKER_STAT_COUNT == cubthread::wp_worker_statset_get_count ());
+  static bool check_names = true;
+  if (check_names)
```

Comment:
Why not copy into perfmon_Portable_worker_stat_names from Worker_pool_statdef instead of checking ?